lib: Know when a process ended due to the child_process.spawn() timeout option #51561

FranciscoMateusVG opened the pull request:
Feature: https://github.com/nodejs/node/issues/51561

Having a way to know whether the process ended due to the timeout option by checking childProcess.timedOut = true

FranciscoMateusVG committed:
lib: add timedOut flag on child process spawn

FranciscoMateusVG committed:
test: add test for checking timedOut flag is set on child process spawn

aduh95 commented:
/cc @nodejs/child_process

sindresorhus commented:
Can someone review?

RafaelGSS reviewed (changes requested):
I'm not on my computer right now - I can check this PR later. But, if you are certain we should have a new property on child, we must document it.

FranciscoMateusVG committed:
doc: add API docs for subprocess.timedOut property

Document the timedOut boolean property on ChildProcess that indicates
whether the child process was killed due to a timeout option.

Refs: https://github.com/nodejs/node/pull/51561

FranciscoMateusVG commented:
@RafaelGSS done!

RafaelGSS reviewed (commented):
CI is red
